sources/common-dylan/byte-vector.dylan

```diff
@@ -221,3 +221,42 @@ define sealed inline method byte-storage-offset-address
            (primitive-repeated-slot-as-raw
               (the-buffer, primitive-repeated-slot-offset(the-buffer)))))
 end method;
+
+define method from-hexstring (string :: <byte-string>)
+  => (result :: <byte-vector>)
+  if (odd?(string.size))
+    error("String size must be multiple of 2.");
+  end if;
+
+  let result :: <byte-vector> = make(<byte-vector>, size: floor/(string.size, 2));
+
+  without-bounds-checks
+    for (i :: <integer> from 0 below string.size - 1 by 2)
```

Comment:
All `floor/(..., 2)` calls in these functions are compiled to shifts:

    t130 := [PRIMOP machine-word-shift-right(t129, ^%2)]

I don't think keeping another index and incrementing it each time will be (significantly) faster.